developers/release.md

```diff
@@ -0,0 +1,235 @@
+
+### Step 3: Setup Your Environment
+
+There are a number of repositories that will need to be tagged and branched throughout the process. We recommend organizing this before hand so that you have a common place to reference and work through these tasks within throughout the release process. Clone the following:
+
+```
+git clone https://github.com/Katello/katello
+git clone https://github.com/Katello/katello-installer.git
+git clone https://github.com/Katello/katello-misc
+git clone https://github.com/Katello/katello-agent
+git clone https://github.com/Katello/hammer-cli-katello
+```
+
+Note that some repositories contain multiple spec-files and thus packages:
+
+```
+katello
+  rubygem-katello.spec
+  deploy/katello.spec
+
+katello-misc
+  certs-tools/katello-certs-tools.spec
+  repos/katello-repos.spec
+  utils/katello-utils.spec
+
+katello-agent
+  katello-agent.spec
+
+katello-installer
+  katello-installer.spec
+
+hammer-cli-katello
+  rubygem-hammer_cli_katello.spec
+```
+
+### Step 4: Get Your Branch On
+
+On the date and time you previously warned the community branching would occur on, do the following for each repository EXCEPT `hammer-cli-katelo`.
+
+The first step is to create the X.Y branch. Take a moment to review the layout of each repository from Step 3. The idea with this step is to checkout the master branch for a given repository, ensure its set to the current upstream master, set a tag point for where you are going to branch for EACH spec file (aka package) that lives in a given repository and then create the branch. For example, the katello repository contains two packages (spec files) at `/` and `/deploy`:
+
+```
+cd katello
+git fetch origin
+git checkout master
+git reset origin/master --hard
+tito tag
+cd deploy/
+tito tag
+git branch -b KATELLO-X.Y
+```
+
+Update the releasers and tito configuration for each package (some repositories contain multiple packages, see Step 3 for list):
+
+```
+vim rel-eng/releasers.conf
+:tabnew rel-eng/tito.props
+
+:tabdo %s/katello-nightly-rhel6/katello-X.Y-rhel6/g
+:tabdo %s/katello-nightly-rhel7/katello-X.Y-rhel7/g
+:wqa
+
+git commit -a -m 'Updating rel-eng for X.Y'
+git push origin KATELLO-X.Y
+```
+
+Now we set master to start from the version after the one we are working on releasing X.Y+1, for each package, ensuring you are in the directory the spec file exists:
+
+```
+git checkout master
+```
+
+For any of the gems, update the gem version file and the spec file to X.Y.0 and make sure to reset the release to 1.
+
+```
+vim lib/katello/version.rb
+vim rubygem-katello.spec
+git commit -a -m 'Version bump to X.Y.0'
+```
+
+Now for all of the repositories including the gems, tag and push the starting place for the next version:
+
+```
+tito tag --use-version=X.Y+1.0
```

Comment:
Why do we do this? Seems weird to already create 2.4.0 *tagged* releases when releasing 2.3.0.  I'm sure there's a good reason, just don't understand it.